### PR TITLE
[release-4.16] OCPBUGS-44485: V1.36.1 rebase release 4.16

### PR DESCRIPTION
**Pull request by arkadeepsen**
Manual cherry-pick of #392, #399, #400 and #402

Followed the below steps:

- Rebase to upstream v1.36.1

  - Ran the following script to create the rebase branch:

    ```
    ./UPSTREAM-MERGE.sh v1.36.1 release-4.16
    ```
Manually cherry-picked the following commits
- Regenerate 03-setversion.patch (262b54113cbd72b525cfdc9f5f2685e111103ef6)
- Regenerate fixsanity patch (fc4703497a0c1166bc73665bf8aa588c79f875c6)
- Regenerate disable security context patch (bcb3bc5b5cc07cbb044a0b03e2a134de14a66a48)
- Regenerate 06-build-darwin-binary.patch (60a012d7fd0983ef26bbf94c19b0bd2a5353a742)
- Regenerate 11-openshiftv1-cli.patch (7c904afcbd49849d0ad49d06d3229880510743b2)
- Remove ansible code (628f54b472ae0b5e4934a9288fb3798b05c89fad)
- Update vendor directory (0e6dbabe79a7817cba432207a183ec22bc643364)
- Set the value of GOBIN in e2e-helm.sh before "make install" of operator-sdk (609dced0b3e363639109865bc6e6e3ce00630fe3)
- Update OCP version (ce6038d0c86034cdf75f3427392f1ba5d7fe9419) and modified OCP version to 4.16
- Update UBI minimal version (9a8f2d8068e4d48de558c7b2ea65b6ade5cd3ab8)
- Remove unused symlink of Dockerfile (53c75adefa78dd5e731ca8fa7c6b3d81b3a58613)
- Update base image of scaffolded Dockerfile for helm-operator to use rhel9 repository (de142a4e0a59fa41f3a219ad8e10c20bf7bdb55d)
- Update base image of scaffolded Dockerfile for ansible-operator to use rhel9 repository (c6597b9b205e621ff0bd5889efac1daf0a63e501)
- Update base image of scaffolded Dockerfile for ose-kube-rbac-proxy to use rhel9 repository (cc8eb91d7766642ae6cf83aab8ec263196b6f9b1)

### CODE REVIEW

File: internal/plugins/openshift/v1/init.go [hunk 1]
```diff
@@ -65,19 +65,19 @@ var imageSubstitutions = map[string][]substitution{
...
-			"registry.redhat.io/openshift4/ose-kube-rbac-proxy:v" + ocpProductVersion,
+			"registry.redhat.io/openshift4/ose-kube-rbac-proxy-rhel9:v" + ocpProductVersion,
 		},
 	},
 	filepath.Join("Dockerfile"): {
 		// Ansible
 		{
 			regexp.MustCompile(`quay.io/operator-framework/ansible-operator:[^ \n]+`),
-			"registry.redhat.io/openshift4/ose-ansible-operator:v" + ocpProductVersion,
+			"registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v" + ocpProductVersion,
```
Comment (KeenonLee) [inline internal/plugins/openshift/v1/init.go:75]:
```
[root@preserve-olm-agent-test ~]# podman pull registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.16
Trying to pull registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.16...
Error: initializing source docker://registry.redhat.io/openshift4/ose-ansible-rhel9-operator:v4.16: reading manifest v4.16 in registry.redhat.io/openshift4/ose-ansible-rhel9-operator: manifest unknown

```

Comment (arkadeepsen) [reply internal/plugins/openshift/v1/init.go:75]: The image will be ready after https://github.com/openshift-eng/ocp-build-data/pull/5665 is merged.

Comment (KeenonLee) [reply internal/plugins/openshift/v1/init.go:75]: Thanks~

**Review by chiragkyal — COMMENTED** [review]
- Image tags for helm operator can be updated to use RHEL9 : `registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16`  , and `4.16:base`
https://github.com/arkadeepsen/ocp-release-operator-sdk/blob/177fa224746c4d35362ea2dd62cdbd23915feb3a/release/helm/Dockerfile#L1 
- I think the ansible folder can be removed completely 
https://github.com/arkadeepsen/ocp-release-operator-sdk/tree/v1.36.1-rebase-release-4.16/release/ansible

**Comment by arkadeepsen** [comment]
> * Image tags for helm operator can be updated to use RHEL9 : `registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16`  , and `4.16:base`
>   https://github.com/arkadeepsen/ocp-release-operator-sdk/blob/177fa224746c4d35362ea2dd62cdbd23915feb3a/release/helm/Dockerfile#L1

There's already a PR for the same. We can get this change from the PR itself: https://github.com/openshift/ocp-release-operator-sdk/pull/372

> * I think the ansible folder can be removed completely
>   https://github.com/arkadeepsen/ocp-release-operator-sdk/tree/v1.36.1-rebase-release-4.16/release/ansible

Thanks for pointing is out. I will remove the folder as well.

**Comment by chiragkyal** [comment]
> There's already a PR for the same. We can get this change from the PR itself: https://github.com/openshift/ocp-release-operator-sdk/pull/372

That PR is changing `go.mod` as well. It should be a conflicting change because we already have `go1.21`.

**Comment by arkadeepsen** [comment]
> > There's already a PR for the same. We can get this change from the PR itself: #372
> 
> That PR is changing `go.mod` as well. It should be a conflicting change because we already have `go1.21`.

Asked in the #forum-ocp-art channel regarding the same: https://redhat-internal.slack.com/archives/CB95J6R4N/p1732090395530879

**Comment by arkadeepsen** [comment]
> > > There's already a PR for the same. We can get this change from the PR itself: #372
> > 
> > 
> > That PR is changing `go.mod` as well. It should be a conflicting change because we already have `go1.21`.
> 
> Asked in the #forum-ocp-art channel regarding the same: https://redhat-internal.slack.com/archives/CB95J6R4N/p1732090395530879

As per the response in the slack thread, we can ignore #372 and create another PR to bump the image versions of helm-operator Dockerfile. I will add those changes as another commit to this PR itself then.

**Comment by chiragkyal** [comment]
/lgtm
Thanks!

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved
/label cherry-pick-approved

**Pull request merged**